src/app/services/courses.service.ts

```diff
@@ -43,47 +47,66 @@ const coursesList = [{
 
 @Injectable()
 export class CoursesService {
-  public getCourses () {
-    return coursesList;
+  private isPopupDisplayed;
+  private deleteCourseId;
+  private courses: BehaviorSubject<ICourse[]> = new BehaviorSubject([]);
+
+  constructor () {
+    this.isPopupDisplayed = false;
+  }
+
+  public get getCourses (): Observable<ICourse[]> {
+    return this.courses;
   }
 
-  public createNewCourse (course) {
+  public getCourseDeleay () {
```

Comment:
generalnie to, że użyłeś `BehaviorSubject` to dobrze, mam tylko zastrzeżenie do nazewnictwa metod ;)
```
getCourses - de facto to jest tylko Observable do ktorego w komponencie dodamy `subscribe`
getCourseDeleay - de facto ta metoda pobiera kolekcje kursów,
```